contracts/eosio.system/src/rex.cpp

```diff
@@ -616,6 +616,69 @@ namespace eosiosystem {
 
    }
 
+   /**
+    * @brief Adds returns from the REX return pool to the REX pool
+    */
+   void system_contract::update_rex_pool()
+   {
+      const time_point ct = current_time_point();
+      const auto ret_pool_elem = _rexretpool.begin();
+
+      if ( ret_pool_elem == _rexretpool.end() ||
+           ct < ret_pool_elem->last_update_time + eosio::microseconds{rex_return_pool::dist_interval}) {
+         return;
+      }
+
+      const int64_t    current_rate    = ret_pool_elem->current_rate_of_increase;
+      const uint64_t   time_interval   = ct.time_since_epoch().count() - ret_pool_elem->last_update_time.time_since_epoch().count();
+      const int64_t    change_estimate = ( uint128_t(time_interval) * current_rate ) / rex_return_pool::total_duration;
```

Comment:
To avoid estimation errors, the `current_rate` (after being updated to include the effects of all buckets with time less than or equal to the current time) must be multiplied by an integer number. In other words take the whole number of `dist_interval` microseconds that has elapsed since the `last_distribution_time` and multiply that by the `current_rate` to determine how much needs to be distributed.

Furthermore, when calculating the distribution rate per bucket, you can calculate the rounding error right there. That error can be distributed immediately to the REX pool. This means that if a small amount of tokens are accumulated during a 12 hour period, 100% of those proceeds may be returned immediately and 0% distributed over 30 days. But for the numbers we have chosen here, it should be a small threshold. For example, with 10 minutes for the distribution interval, and 0.0001 SYS as the smallest non-zero amount of system tokens, we would only have 100% of proceeds distributed immediately if less than 0.4320 SYS was collected in proceeds within a 12 hour bucket. However, if, for example, 10 SYS was collected in proceeds within a 12 hour bucket, then 0.0023 SYS would be distributed every 12 hours over the course of 30 days and the residual amount of 0.0640 SYS due to rounding errors (which is less than 0.1% of the total proceeds collected in the 12 hour bucket) would be distributed immediately to the REX pool after the 12 hour bucket time window passes.